File: JAVAapp/SharedPreferences2/app/src/main/java/org/techtown/sharedpreferences2/MainActivity.java
package org.techtown.sharedpreferences2;

import android.app.Activity;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    SharedPreferences pref;          // 프리퍼런스
    SharedPreferences.Editor editor; // 에디터

    int myInt;                      // 숫자 변수
    String myStr;                   // 문자 변수

    EditText et01, et02;            // 레이아웃 변수 EditText
    Button btn01, btn02;            // 레이아웃 변수 Button

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // 1. Shared Preference 초기화
        pref = getSharedPreferences("pref", Activity.MODE_PRIVATE);
        editor = pref.edit();

        // 2. 저장해둔 값 불러오기 ("식별값", 초기값) -> 식별값과 초기값은 직접 원하는 이름과 값으로 작성.
        myInt = pref.getInt("MyInt", 0);        // int 불러오기 (저장해둔 값 없으면 초기값인 0으로 불러옴)
        myStr = pref.getString("MyStr", "_");   // String 불러오기 (저장해둔 값 없으면 초기값인 _으로 불러옴)


        // 3. 레이아웃 변수 초기화
        et01 = findViewById(R.id.et01); et02 = findViewById(R.id.et02);
        btn01 = findViewById(R.id.btn01); btn02 = findViewById(R.id.btn02);

        // 4. 앱을 새로 켜면 이전에 저장해둔 값이 표시됨
        et01.setText(String.valueOf(myInt)); et02.setText(myStr);


        // 5. 각 버튼 클릭시 새로운 값 저장
        btn01.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                myInt = Integer.parseInt(et01.getText().toString()); // int Max값 넘게 입력하면 오류 주의.
                editor.putInt("MyInt", myInt);
                editor.apply(); // 저장
            }
        });
        btn02.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                myStr = et02.getText().toString();
                editor.putString("MyStr", myStr);
                editor.apply(); // 저장
            }
        });
    }
}

File: KotlinApp/Lifecycle_Test/app/src/main/java/Activity/MainActivity.kt
package Activity

import android.os.Bundle
import android.os.PersistableBundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import org.techtown.lifecycle_test.R

class MainActivity : AppCompatActivity() {
    // 최초로 Activity가 생성 될 때 실행되는 메서드 단 한 번만 호출 된다.
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        Log.d("Life_cyle", "onCreate")
    }

    override fun onRestart() {
        super.onRestart()
        Log.d("Life_cyle", "onRestart")
    }

    // onCreate() 이후에 실행 된다 Activity가 실행되며 사용자가 앱과 상호 작용을 준비한다.
    override fun onStart() {
        super.onStart()
        Log.d("Life_cyle", "onStart")
    }

    //     다시 앱으로 돌아올 때 호출 된다.
    override fun onResume() {
        super.onResume()
        Log.d("Life_cyle", "onResume")
    }

    //     화면의 일부가 가려 졌을 떄
    override fun onPause() {
        super.onPause()
        Log.d("Life_cyle", "onPause")
    }

    //    화면 전부가 보이지 않을 때때
    override fun onStop() {
        super.onStop()
        Log.d("Life_cyle", "onStop")
    }

    //    완전히 소멸 되기전에 실행
    override fun onDestroy() {
        super.onDestroy()
        Log.d("Life_cyle", "onDestroy")
    }
}